torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -39,7 +39,16 @@ def decorator(wrapper_factory):
 class VisionDatasetDatapointWrapper(Dataset):
     def __init__(self, dataset):
         dataset_cls = type(dataset)
+        # We test for exact dataset class matches first. If we don't find one, we check if the dataset subclasses any
+        # of the known ones.
         wrapper_factory = WRAPPER_FACTORIES.get(dataset_cls)
+        if wrapper_factory is None:
+            with contextlib.suppress(StopIteration):
+                wrapper_factory = next(
+                    wrapper_factory
+                    for dataset_supercls_candidate, wrapper_factory in WRAPPER_FACTORIES.items()
+                    if issubclass(dataset_cls, dataset_supercls_candidate)
+                )
```

Comment:
IIUC we're chosing the first wrapper in `WRAPPER_FACTORIES` that is associated with a subclass of the dataset.

Isn't that a problem if e.g. `ImageFolder` is one of the first entries in WRAPPER_FACTORIES (assuming we have a wrapper for this one?)

It feels like we should go up the MRO instead and look for an exact match in the WRAPPER_FACTORIES. But... maybe I'm overthinking. I'll let you decide - LGTM with some small unit tests :)

Reply:
Fair enough, that seems like a safer method to do that. Thanks!